packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -5837,12 +5833,14 @@ namespace Tpetra {
       }
       else {
         size_t check_row_length = 0;
-        src_row_inds.resize(num_src_entries);
-        source.getGlobalRowCopy(src_gid, src_row_inds(), check_row_length);
+        std::vector<GO> src_row_inds(num_src_entries);
```

Comment:
Tim and I talked on the phone about this.  As far as he can tell, some issue with Teuchos memory management classes' debug-mode checking caused the test failures.  That's why he replaced the `Teuchos::Array` with `std::vector` here.

@tjfulle , you could use exactly the same `resize` logic for `std::vector` as for `Teuchos::Array`; it should work fine.  In fact, you don't have to resize the vector if it already has enough space; just resize if you need more space, and always give `Teuchos::ArrayView`'s constructor `num_src_entries` instead of `src_row_inds.size()` as its size.